go/libraries/doltcore/schema/check_coll.go

```diff
@@ -33,6 +33,8 @@ type CheckCollection interface {
 	DropCheck(name string) error
 	// AllChecks returns all the checks in the collection
 	AllChecks() []Check
+	// Equals returns whether the provided check collection is equal or not.
+	Equals(other CheckCollection) bool
```

Comment:
there's really no reason for `CheckCollection` to be an interface, but that's for another day